packages/alfa-dom/src/common/predicate.ts

```diff
@@ -0,0 +1,32 @@
+import { Predicate } from "@siteimprove/alfa-predicate";
+import {Element, Namespace, Node} from "..";
+
+const { and, equals, property } = Predicate;
+
+export function hasNamespace(
+  predicate: Predicate<Namespace> = () => true
+): Predicate<Element> {
+  return (element) => element.namespace.some(predicate);
+}
+
+export function hasName<T extends { readonly name: string }>(
+  predicate: Predicate<string> = () => true
+): Predicate<T> {
+  return property("name", predicate);
+}
```

Comment:
This predicate is not related to DOM as it's just a wrapper around `property("name", predicate)` so let's not put it in the DOM package. I'm thinking `Element#hasName(predicate: Predicate<string>): boolean` and `Element.hasName(predicate: Predicate<string>): Predicate<Element>` would make sense though. This should also remove the need for `isElementByName()`.